src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -1034,1026 +1479,2694 @@ static void Main()
+                  IL_0006:  dup
+                  IL_0007:  ldc.i4.1
+                  IL_0008:  ldc.i4.7
+                  IL_0009:  conv.i8
+                  IL_000a:  newobj     "long?..ctor(long)"
+                  IL_000f:  stelem     "long?"
+                  IL_0014:  newobj     "System.{{spanType}}<long?>..ctor(long?[])"
+                  IL_0019:  ret
+                }
+                """);
+        }
+
+        [ConditionalTheory(typeof(CoreClrOnly))]
+        [CombinatorialData]
+        public void Span_02(bool useReadOnlySpan)
+        {
+            string spanType = useReadOnlySpan ? "ReadOnlySpan" : "Span";
+            string source = $$"""
+                using System;
+                class Program
+                {
+                    static void Main()
+                    {
+                        {{spanType}}<string> x = [];
+                        {{spanType}}<int> y = [1, 2, 3];
+                        x.Report();
+                        y.Report();
+                    }
+                }
+                """;
+            CompileAndVerify(new[] { source, s_collectionExtensionsWithSpan }, targetFramework: TargetFramework.Net70, verify: Verification.Skipped, expectedOutput: "[], [1, 2, 3], ");
+        }
+
+        [ConditionalTheory(typeof(CoreClrOnly))]
+        [CombinatorialData]
+        public void Span_03(bool useReadOnlySpan)
+        {
+            string spanType = useReadOnlySpan ? "ReadOnlySpan" : "Span";
+            string source = $$"""
+                using System;
+                class Program
+                {
+                    static void Main()
+                    {
+                        var x = ({{spanType}}<string>)[];
+                        var y = ({{spanType}}<int>)[1, 2, 3];
+                        x.Report();
+                        y.Report();
+                    }
+                }
+                """;
+            CompileAndVerify(new[] { source, s_collectionExtensionsWithSpan }, targetFramework: TargetFramework.Net70, verify: Verification.Skipped, expectedOutput: "[], [1, 2, 3], ");
+        }
+
+        [ConditionalTheory(typeof(CoreClrOnly))]
+        [CombinatorialData]
+        public void Span_04(bool useReadOnlySpan)
+        {
+            string spanType = useReadOnlySpan ? "ReadOnlySpan" : "Span";
+            string source = $$"""
+                using System;
+                class Program
+                {
+                    static ref readonly {{spanType}}<int> F1()
+                    {
+                        return ref F2<int>([]);
+                    }
+                    static ref readonly {{spanType}}<T> F2<T>(in {{spanType}}<T> s)
+                    {
+                        return ref s;
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyEmitDiagnostics(
```

Comment:
Do we really need Emit diagnostics to verify this? Please prefer `VerifyDiagnostics` over `VerifyEmitDiagnostics` for positive diagnostics scenarios, as that conveys information about whether the errors will reported live in the IDE, or if they will only appear during a full build.